### PR TITLE
Coaches migration model

### PR DESCRIPTION
**Pull request by chrisoney**

### CODE REVIEW

File: app/models/api/coach.rb [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+#
+#  id              :bigint           not null, primary key
+#  first_name      :string           not null
+#  last_name       :string           not null
+#  email           :string           not null
+#  password_digest :string           not null
+#  created_at      :datetime         not null
+#  updated_at      :datetime         not null
+#
+class Coach < ApplicationRecord
```
Comment (rweald) [inline app/models/api/coach.rb:13]: We don't really need to nest this model in the `/api` folder. Generally speaking I tend to only really like nesting models once there are clearly defined domain concepts that can act as boundaries. Don't think we have those yet, and even if we had a sense for some, like perhaps Messaging or SMS, the coach in particular is definitely a broadly applicable model.

File: app/models/api/coach.rb [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+  attr_reader :password
+
+  validates :first_name, presence: true
+  validates :last_name, presence: true
+  validates :email, presence: true
+  validates :password_digest, presence: true
+  validates :password, length: { minimum: 6 }, allow_nil: true
+
+  def password=(password)
```
Comment (rweald) [inline app/models/api/coach.rb:22]: Don't do the password related stuff yourself. Use `has_secure_password` then rails handles everything for you: validations, hashing, even auth checks.

File: app/models/api/coach.rb [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+#  password_digest :string           not null
+#  created_at      :datetime         not null
+#  updated_at      :datetime         not null
+#
+class Coach < ApplicationRecord
+  attr_reader :password
+
+  validates :first_name, presence: true
+  validates :last_name, presence: true
+  validates :email, presence: true
```
Comment (rweald) [inline app/models/api/coach.rb:18]: We need to ensure email is unique in order for auth to use email as the identity selector. When adding the uniqueness validations in the app layer, also make sure to add a unique index via migration.

**Review by rweald — COMMENTED** [review]
Let's add an associated spec file to test the validations you enabled on the coach object. Super easy to test validations using [shoulda-matchers](https://github.com/thoughtbot/shoulda-matchers) which I've already configured in the project. 

You can do something like: 

```
describe "validations" do 
    it { is_expected.to validate_presence_of(:first_name) }
    it { is_expected.to validate_presence_of(:last_name) }
    it { is_expected.to validate_presence_of(:email) }
    ...
```

**Comment by rweald** [comment]
We should also add a valid Factory for generating Coaches in any subsequent testing.

**Comment by rweald** [comment]
@chrisoney this is what my PR for a similar change / feature set looks like - https://github.com/coachmehealth/sms-care-portal-rails/pull/6. See how it includes the specs, factory setup, and model layering.

**Comment by chrisoney** [comment]
Thank you for reviewing this so quickly and for giving such great feedback. I'll make the updates. Would you prefer I update the pull request or just do a new one entirely?

**Pull request closed without merging**